common/src/com/thoughtworks/go/remote/work/ConsoleOutputTransmitter.java

```diff
@@ -86,10 +94,9 @@ public void flushToServer() {
             consoleAppender.append(result.toString());
         } catch (IOException e) {
             LOGGER.warn("Could not send console output to server", e);
-            //recreate buffer
             synchronized (buffer) {
                 sent.addAll(buffer);
-                buffer = new CircularFifoBuffer(10 * 1024);
+                buffer.clear();
```

Comment:
This change makes sense to me too. Its ok, ignore it for now. This would be dog-fooded to a few environments so we should get some feedback, lets just keep it at the back of our heads.